src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1599,20 +1599,12 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
                     // This is not a problem, because the argument explicitly
                     // requests constness, in contrast to regular promotion
                     // which happens even without the user requesting it.
-                    // We can error out with a hard error if the argument is not
-                    // constant here.
+                    //
+                    // `promote_consts` is responsible for emitting the error if
+                    // the argument is not promotable.
                     if !IsNotPromotable::in_operand(self, arg) {
                         debug!("visit_terminator_kind: candidate={:?}", candidate);
                         self.promotion_candidates.push(candidate);
-                    } else {
-                        if is_shuffle {
-                            span_err!(self.tcx.sess, self.span, E0526,
-                                      "shuffle indices are not constant");
```

Comment:
Probably need to fix tests because this error isn't emitted anymore?
I wonder what @gnzlbg thinks (it wouldn't be that hard to keep tbh).

Reply:
There are no tests for non-const `simd_shuffle` args. I could add one in this PR

Reply:
I can store `is_shuffle` in `Candidate::Argument` for now.

Reply:
I'm not actually sure how to test `simd_shuffle`, I might have to add a test in `stdarch`? The only thing exposed from `core` are platform-specific intrinsics that wrap the LLVM one and that already have `rustc_args_required_const`. Let me know if you have ideas.

Reply:
> Probably need to fix tests because this shuffle-specific error message/code isn't emitted anymore? (i.e. now shuffles use the more general error)
I wonder what @gnzlbg thinks (it wouldn't be that hard to keep tbh).

The shuffle intrinsics are only directly used by libcore, so this errors should only be seen by `libcore`-developers. I don't think having an specific error for this is worth it, the general one should do. https://github.com/rust-lang/stdarch/pull/825 looks like a nice way to solve this problem, so this LGTM.